x-pack/otel/processor/beatprocessor/integration_test.go

```diff
@@ -0,0 +1,292 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package beatprocessor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/otelbeat/oteltest"
+	libbeattesting "github.com/elastic/beats/v7/libbeat/testing"
+	"github.com/elastic/beats/v7/libbeat/tests/integration"
+	"github.com/elastic/elastic-agent-libs/testing/estools"
+
+	"github.com/gofrs/uuid/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOtelBeatProcessorE2E(t *testing.T) {
+	integration.EnsureESIsRunning(t)
+	wantEvents := 1
+
+	// Start Filebeat receiver with Beat processors in a separate OTel processor
+	processorFilebeat := integration.NewBeat(
+		t,
+		"filebeat-otel",
+		"../../../filebeat/filebeat.test",
+		"otel",
+	)
+
+	namespace := strings.ReplaceAll(uuid.Must(uuid.NewV4()).String(), "-", "")
+	processorIndex := "logs-processor-" + namespace
+	receiverIndex := "logs-receiver-" + namespace
+
+	processor := int(libbeattesting.MustAvailableTCP4Port(t))
+	receiverMonitoringPort := int(libbeattesting.MustAvailableTCP4Port(t))
```

Comment:
You can probably omit monitoring, it is not needed to prove the beat processor is working and makes the test simpler.